plugins/woocommerce/tests/e2e-pw/tests/merchant/order-status-filter.spec.js

```diff
@@ -55,26 +56,15 @@ test.describe(
 			await api.post( 'orders/batch', { delete: [ ...orderBatchId ] } );
 		} );
 
-		test( 'should filter by All', async ( { page } ) => {
-			await page.goto( '/wp-admin/admin.php?page=wc-orders' );
-
-			await page.locator( 'li.all > a' ).click();
-			// because tests are running in parallel, we can't know how many orders there
-			// are beyond the ones we created here.
-			for ( let i = 0; i < orderStatus.length; i++ ) {
-				const statusTag = 'text=' + orderStatus[ i ][ 0 ];
-				const countElements = await page.locator( statusTag ).count();
-				await expect( countElements ).toBeGreaterThan( 0 );
-			}
-		} );
-
 		for ( let i = 0; i < orderStatus.length; i++ ) {
 			test( `should filter by ${ orderStatus[ i ][ 0 ] }`, async ( {
 				page,
 			} ) => {
 				await page.goto( '/wp-admin/admin.php?page=wc-orders' );
 
 				await page.locator( `li.${ orderStatus[ i ][ 1 ] }` ).click();
+				// eslint-disable-next-line playwright/no-networkidle
+				await page.waitForLoadState( 'networkidle' );
 				const countElements = await page
```

Comment:
Against Pressable, the tests were failing between `await page.locator( 'li.${ orderStatus[ i ][ 1 ] }' ).click();` and `const countElements = await page.locator( statusColumnTextSelector ).count();`. I'm not sure why, but `countElements` was always `0`. I solved this with `networkidle`, and tests are passing now everytime, but if there's a better way to wait between these two actions please let me know.

Reply:
Try using [toHaveCount](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count) assertion instead, which will wait until the number is the expected one or timeout. Using `count` will not wait, and on slower sites (like I've seen Pressable to be) it makes sense to fail. I'm surprised this is not failing more often TBH.

Reply:
I addressed it here [3012d88](https://github.com/woocommerce/woocommerce/pull/51158/commits/3012d884f056e57e182e9efd191039b7777c3eae)

Do you think it's a viable solution or should I proceed with [toHaveCount](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count)?

BTW, thanks, TIP about [toHaveCount](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count) 🙌

Reply:
I think [toHaveCount](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count) should be used in this case anyway. It's the proper way to assert a count with Playwright.